src/GeneratedMapper/SyntaxReceivers/MapAttributeReceiver.cs

```diff
@@ -9,6 +10,9 @@ namespace GeneratedMapper.SyntaxReceivers
     internal sealed class MapAttributeReceiver : ISyntaxReceiver
     {
         public List<TypeDeclarationSyntax> Candidates { get; } = new List<TypeDeclarationSyntax>();
+        private List<TypeDeclarationSyntax> _cachedSyntaxes = new List<TypeDeclarationSyntax>();
+        public List<Tuple<TypeDeclarationSyntax, TypeDeclarationSyntax>> ExtensionCandidates { get; } = new List<Tuple<TypeDeclarationSyntax, TypeDeclarationSyntax>>();
+        private List<Tuple<TypeSyntax, TypeSyntax>> _extensionCandidates = new List<Tuple<TypeSyntax, TypeSyntax>>();
```

Comment:
I think its better to use value types instead of `Tuple<>`, so `List<(TypeSyntax source, TypeSyntax destination)>`

Reply:
Agreed.  This was simple quick and dirty way of getting it to work.
I might have to throw in SyntaxReference object in the `ExtensionCandiates` so I can know where the source is, since it's not always an attribute that is the source of the mapping code generation.

Reply:
Did make ExtensionMapCandidate to fix this issue already.